### PR TITLE
Add UART example.

### PR DESCRIPTION
**Pull request by floitsch**

### CODE REVIEW

File: uart/README.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
+# UART Example
+
+Demonstrates the use of the UART.
+
+## How to run
+
+This demo can be run on one or two ESP32 boards.
+It uses: RX=16, TX=17 for one UART and RX=33, TX=32 for the other.
```
Comment (andersjohnsen) [inline uart/README.md:8]: Let's use pins that are not occupied on PSRAM devices, e.g. 21 and 22.

File: uart/ping.toit [hunk 2]
```diff
@@ -0,0 +1,41 @@
+// Copyright (C) 2021 Toitware ApS.
+// Use of this source code is governed by a Zero-Clause BSD license that can
+// be found in the EXAMPLES_LICENSE file.
+
+import uart
```
Comment (andersjohnsen) [inline uart/ping.toit:5]: IMO this is slightly more complicated then needed. You can have a single UART echo to itself, by using pin 21 as TX and 22 as RX, and wire those two together. That way you don't need multiple UARTs running.

File: uart/pong.toit [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+import uart
+import writer show Writer
+import gpio
+
+RX ::= 33
+TX ::= 32
+
+main:
+  // The UART port can be configured in many ways.
+  // See https://libs.toit.io/uart/class-Port#constructor(10%2C0%2C0%2Cbaud_rate%2Ccts%2Cdata_bits%2Cinvert_rx%2Cinvert_tx%2Cparity%2Crts%2Crx%2Cstop_bits%2Ctx)
```
Comment (andersjohnsen) [inline uart/pong.toit:14]: Any chance we can use a link that doesn't include the signature? This looks very fragile to API extentions.

**Review by andersjohnsen — APPROVED** [review]

File: uart/ping.toit (hunk 2 above)
Comment (floitsch) [reply uart/ping.toit:5]: I want to show a case where we use a full UART in an app.
If we only used the RX (resp. TX), then that would only show half of it.

File: uart/pong.toit (hunk 3 above)
Comment (floitsch) [reply uart/pong.toit]: Currently it's not possible.
Linking just to the 'Port' class.

File: uart/ping.toit (hunk 2 above)
Comment (andersjohnsen) [reply uart/ping.toit:5]: You'd still do this:
```
main:
  port := uart.....
  
  task::
    while line := reader.read_line:
      print line
      
  10.repeat:
    port.write "lala $it"
    sleep --ms=1000
```

I still believe this example is much easier to understand.

Comment (floitsch) [reply uart/ping.toit:5]: Ok.
The main advantage of the current example is, that I can also show the `data := port.read`.
Will think how we can make things better. (Don't hesitate to ping me so I don't forget).

**Pull request merged**